src/utils/EventUtils.js

```
@@ -0,0 +1,22 @@
+/**
+ * Chain event handlers.
+ *
+ * Returns an event handler which:
+ * - calls all the given event handler by order
+ * - If an exception is thrown in the a handler, then the rest of the handler will NOT be called.
+ * - returns the return value of the first (defined) handler
+ * @returns An event handler that chains all given handlers
+ */
+export function chainEventHandlers(...handlers) {
+  return event => {
+    let res;
+    handlers.forEach(h => {
+      if (h) {
+        const r = h && h(event);
```

no need to check for `h` twice

im also thinking perhaps the return value of handler is not needed? here
the last result is returned so the others are discarded anyway. if so, it can become simpler

```js
export const chainEventHandlers = (...handlers) =>
  event =>
    handlers
      .filter(i => i)
      .forEach(handler => handler(event))
```

i would also put this into dedicated file `chain-event-handlers.js` but that's not really important

---

i don't know. maybe if some handler fails, let it crash? otherwise you'd need some error message, explanation and stuff. people might argue why we don't do defenses everywhere then. but i don't know, we can flip a coin :)